tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1576 @@
+        f"""
+           class_name: ConfiguredAssetAzureDataConnector
+           datasource_name: test_environment
+           execution_engine:
+               class_name: PandasExecutionEngine
+           container: my_container
+           name_starts_with: ""
+           assets:
+               TestFiles:
+           default_regex:
+               pattern: (.+)_(.+)_(.+)\\.csv
+               group_names:
+                   - name
+                   - timestamp
+                   - price
+           azure_options:
+               account_url: my_account_url.blob.core.windows.net
+               credential: my_credential
+       """,
+    )
+
+    mock_list_keys.return_value = (
+        [
+            "alex_20200809_1000.csv",
+            "eugene_20200809_1500.csv",
+            "james_20200811_1009.csv",
+            "abe_20200809_1040.csv",
+            "will_20200809_1002.csv",
+            "james_20200713_1567.csv",
+            "eugene_20201129_1900.csv",
+            "will_20200810_1001.csv",
+            "james_20200810_1003.csv",
+            "alex_20200819_1300.csv",
+        ],
+    )
+
+    my_data_connector: ConfiguredAssetAzureDataConnector = (
+        instantiate_class_from_config(
+            config=my_data_connector_yaml,
+            runtime_environment={
+                "name": "general_azure_data_connector",
+                "datasource_name": "test_environment",
+            },
+            config_defaults={
+                "module_name": "great_expectations.datasource.data_connector"
+            },
+        )
+    )
+
+    with pytest.raises(TypeError):
+        my_data_connector.get_batch_definition_list_from_batch_request()
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_get_definition_list_from_batch_request_with_unnamed_data_asset_name_raises_error(
+    mock_azure_conn, mock_list_keys, mock_emit, empty_data_context_stats_enabled
+):
+    my_data_connector_yaml = yaml.load(
+        f"""
+           class_name: ConfiguredAssetAzureDataConnector
+           datasource_name: test_environment
+           execution_engine:
+               class_name: PandasExecutionEngine
+           container: my_container
+           name_starts_with: ""
+           assets:
+               TestFiles:
+           default_regex:
+               pattern: (.+)_(.+)_(.+)\\.csv
+               group_names:
+                   - name
+                   - timestamp
+                   - price
+           azure_options:
+               account_url: my_account_url.blob.core.windows.net
+               credential: my_credential
+       """,
+    )
+
+    mock_list_keys.return_value = (
```

Comment:
Is this needed, given that we will be causing an error to be raised (by specifying `data_asset_name=None`)?

Reply:
That's a great point and one that I overlooked 👀 

Made the update!